### PR TITLE
Feat/Holders insider data

### LINKED ISSUE #1529: Add method for insider roster
**Describe the problem**
There currently exists no method to access the insider roster available on the site, next to the holders, e.g. https://finance.yahoo.com/quote/MSFT/insider-roster?p=MSFT

**Describe the solution**
Like the Holders class, implement one for InsiderRoster. 

**Additional context**

Here is how I am hacking it for now:

```python
def get_insider_roster(ticker: yf.Ticker, proxy = None) -> pd.DataFrame:
    url = f'https://finance.yahoo.com/quote/{ticker._data.ticker}/insider-roster'
    resp = ticker._data.cache_get(url, proxy)
    insiders = pd.read_html(resp.text)
    insiders = insiders[0].iloc[:-1]
    return insiders
```

### PR DESCRIPTION
**Pull request by JuliaLWang8**
Resolves #1529: Added implementations for all data displayed on the [holders page](https://finance.yahoo.com/quote/AAPL/holders?p=AAPL). 

Can access dataframes: 
```py
msft = yf.Ticker("MSFT")
msft.insider_transactions
msft.insider_purchases
msft.insider_roster_holders
```

Also added an additional column called 'Position' separating people's names and titles since they are grouped together on the endpoint. This is because Yahoo displays insider names with their titles/positions below their name.

### CODE REVIEW

**Comment by ValueRaider** [comment]
I see opportunities to improve formatting. Data types, index, etc. See `Ticker.get_earnings_dates` for inspiration.

**Comment by sshcli** [comment]
I love this PR, I'm excited just with the idea of see this one included in 0.2.34
Thank you @eshgovil for creating the issue and put it into the radar.
Thank you @ValueRaider for being so nice by encouraging others to participate and collaborate in this amazing project.
Thank you @JuliaLWang8 for being so diligent and proactive.

File: yfinance/scrapers/holders.py [hunk 1]
```diff
@@ -34,6 +38,24 @@ def mutualfund(self) -> pd.DataFrame:
...
+    def insider_purchases(self) -> pd.DataFrame:
+        if self._insider_purchases is None:
+            self._scrape_insider_transactions(self.proxy)
+        return self._insider_purchases
+    
+    @property
+    def insider_roster(self) -> pd.DataFrame:
+        if self._insider_roster is None:
+            self._scrape_insider_ros(self.proxy)
+        return self._insider_roster
```
[the remark this answers is not in the document]
Comment (JuliaLWang8) [reply yfinance/scrapers/holders.py:59]: > I don't known how to pull from pr to here. So i create other [pull](https://github.com/ranaroussi/yfinance/pull/1777) request. Or help me

The only way that I know of for you to contribute to this PR is as a collaborator on my forked branch [JuliaLWang8:feat/holders-insiders](https://github.com/JuliaLWang8/yfinance/tree/feat/holders-insiders) and vice versa for me to make changes on your branch. I've added you as a collaborator to my fork, but it's your choice whichever branch you want to work on.

[the remark this answers is not in the document]
Comment (JuliaLWang8) [reply yfinance/scrapers/holders.py:70]: Works well! I had an issue with the insider rosters dates but I fixed it in my latest commit. I also changed the df output for the purchases to match [Yahoo's page](https://finance.yahoo.com/quote/MSFT/insider-transactions?p=MSFT).

**Comment by JuliaLWang8** [comment]
> Good Job. I think we can merge it.

Sounds good! Could you approve the PR and merge @bot-unit? I don't have access. Or maybe @ValueRaider

File: yfinance/scrapers/holders.py [hunk 2]
```diff
@@ -14,27 +18,198 @@ def __init__(self, data: YfData, symbol: str, proxy=None):
...
+            }, inplace=True)
+
+            df["Name"] = df["Name"].astype(str)
+            df["Position"] = df["Position"].astype(str)
+            df["URL"] = df["URL"].astype(str)
+            df["Most Recent Transaction"] = df["Most Recent Transaction"].astype(str)
+
+        self._insider_roster = df
+
+    def _parse_net_share_purchase_activity(self, data):
```
[the remark this answers is not in the document]
Comment (JuliaLWang8) [reply yfinance/scrapers/holders.py:196]: Done in latest commit!

**Comment by sshcli** [comment]
OMG!
You guys are the best. 
This is what I call a beautiful teamwork! 💪

**Comment by ValueRaider** [comment]
I spotted the `DataFrame.apply()` pattern. If you got access to GPT4, is easy to convert to vectorized equivalents. Usually less and cleaner code too.
```
Hey GPT. Carefully review the following Python Pandas code, and replace nonvectorized operations with vectorized equivalents where possible.
...
```

**Comment by JuliaLWang8** [comment]
> I spotted the `DataFrame.apply()` pattern. 

Removed in last commit @ValueRaider

**Comment by ValueRaider** [comment]
Code looks fine.

Add this to README:
```
msft.insider_transactions
msft.insider_purchases
msft.insider_roster_holders
```

Then squash the commits.

**Comment by JuliaLWang8** [comment]
> Then squash the commits.

Done both!

**Comment by sshcli** [comment]
@ValueRaider @JuliaLWang8 @bot-unit is there a way to pip install` yfinance dev` and keep it updated at all times?
I would like to report any bug I may find in dev environment before they go to production, and at the same time take advantage of new features that are only available in dev.

**Comment by ValueRaider** [comment]
> pip install

#1080

> keep it updated 

No, have to manually reinstall. Subscribe for notifications.

**Comment by sshcli** [comment]
Thanks @ValueRaider

**Pull request merged**